sentry/src/main/java/io/sentry/cache/IEnvelopeCache.java

```diff
@@ -6,10 +6,17 @@
 
 public interface IEnvelopeCache extends Iterable<SentryEnvelope> {
 
+  @Deprecated
   void store(@NotNull SentryEnvelope envelope, @NotNull Hint hint);
 
+  default boolean storeEnvelope(@NotNull SentryEnvelope envelope, @NotNull Hint hint) {
+    store(envelope, hint);
+    return true;
```

Comment:
I think we should actually return `false` in case an envelope couldn't be stored [here](https://github.com/getsentry/sentry-java/blob/6fc8adc22cd3e52adb41031f32c95019de0a2fd0/sentry/src/main/java/io/sentry/cache/EnvelopeCache.java#L311-L315), wdyt? Could be because of no storage space, for example.